### PR TITLE
MGMT-11290: run conformance suite in vsphere flow

### PR DESCRIPTION
**Pull request by osherdp**
Running the whole suite of conformance for the vsphere periodic job.

### CODE REVIEW

**Comment by vrutkovs** [comment]
Check out https://github.com/openshift/release/pull/31144 - it has two commits to replace vSphere stubs with real values (as user is supposed to do post install) and has passing rehearsals

**Comment by osherdp** [comment]
> 

sounds great!
but can you point me to the passing rehearsals? I'm not sure I can find one that actually ran the whole suite and succeeded (although passing more tests than this PR managed to is also an important progress)

**Comment by osherdp** [comment]
Either way I'm amending the changes to be able to test them

**Comment by vrutkovs** [comment]
https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/31144/rehearse-31144-periodic-ci-openshift-assisted-test-infra-master-e2e-vsphere-assisted-periodic/1557740996324757504 has vsphere specific tests, but still needs fixes (most likely StorageClass replacement I guess)

File: ci-operator/step-registry/assisted/common/verify/assisted-common-verify-commands.sh [hunk 1]
```diff
@@ -3,50 +3,41 @@
...
+        exit 1
+        ;;
+esac
+
+for kubeconfig in ${SHARED_DIR}/kubeconfigs/*; do
+    export KUBECONFIG=${kubeconfig}
+    name=$(basename ${kubeconfig})
+
+    openshift-tests run "openshift/conformance/parallel" --dry-run | \
+        grep -Ff /tmp/test-list | \
```
Comment (vrutkovs) [inline ci-operator/step-registry/assisted/common/verify/assisted-common-verify-commands.sh:41]:
```suggestion
        grep -Ff ${SHARED_DIR}/test-list | \
```

Comment (osherdp) [reply ci-operator/step-registry/assisted/common/verify/assisted-common-verify-commands.sh:41]: :facepalm: 
thank you!

File: ci-operator/step-registry/assisted/common/verify/assisted-common-verify-commands.sh [hunk 2]
```diff
@@ -3,50 +3,43 @@
...
+        ;;
+
+    *)
+        echo >&2 "Unsupported cluster type '${CLUSTER_TYPE}'"
+        exit 1
+        ;;
+esac
+
+for kubeconfig in ${SHARED_DIR}/kubeconfigs/*; do
+    export KUBECONFIG=${kubeconfig}
```
Comment (vrutkovs) [inline ci-operator/step-registry/assisted/common/verify/assisted-common-verify-commands.sh:37]: A lot of tests failed with
```
{  error: error loading KubeConfig: open /tmp/secret/kubeconfigs/*: no such file or directory}
```
I think we should use `find -type f` trick here

Comment (vrutkovs) [reply ci-operator/step-registry/assisted/common/verify/assisted-common-verify-commands.sh:37]: Also, isn't it already saved in `${SHARED_DIR}/kubeconfig`?

Comment (osherdp) [reply ci-operator/step-registry/assisted/common/verify/assisted-common-verify-commands.sh:37]: Good idea! ``find -type f`` it is

For some reason the intention in this step is supporting running conformance tests for multiple installed clusters. I do know that we have a couple of jobs doing several installations, so even though I don't like that design I'm neglected from changing it for now
I'm assuming ``${SHARED_DIR}/kubeconfig`` to have just one cluster kubeconfig if I understand the code correctly

**Comment by osherdp** [comment]
Wow, conformance ran for 1h40m until it got interrupted

**Comment by vrutkovs** [comment]
One rehersal passed (:tada:), the other one timed out. Lets merge it and check stats after a few periodic runs
/lgtm

Feel free to pull the hold when ready

**Comment by osherdp** [comment]
> One rehersal passed (tada), the other one timed out. Lets merge it and check stats after a few periodic runs /lgtm
> 
> Feel free to pull the hold when ready

thanks! I just added a shellcheck ignore to fix the job as well

**Pull request merged**